bioptim/gui/graph.py

```diff
@@ -0,0 +1,488 @@
+from graphviz import Digraph
+
+from ..limits.constraints import Constraint
+from ..limits.objective_functions import ObjectiveFcn, ObjectiveList
+from ..optimization.parameters import Parameter
+
+
+class GraphAbstract:
+    _return_line: ""
+    _squared: ""
+
+    def __init__(
+        self,
+        ocp,
+    ):
+        self.ocp = ocp
+
+    def _add_dict_to_str(self, _dict: dict):
+        str_to_add = ""
+        for d in _dict:
+            str_to_add += f"{d}: {_dict[d]}{self._return_line}"
+        return str_to_add
+
+    def _constraint_to_str(self, constraint: Constraint):
+        """
+        Convert constraint information into an easy-to-read string
+
+        Parameters
+        ----------
+        constraint: Constraint
+            The constraint to be converted
+        """
+
+        constraint_str = ""
+        target_str = "" if constraint.sliced_target is None else f"{constraint.sliced_target}"
+        if constraint.quadratic:
+            constraint_str += f"{constraint.min_bound} ≤ "
+            constraint_str += f"({constraint.name}" if target_str is not "" else f"{constraint.name}"
```

Comment:
Watch out with "is not". Is not refer to the type and not to the instance itself. In practice, this will work, but is considered bad practice since Python 3.7. The expected code here is: `target_str != ""`
Please replace them in the whole file

Reply:
Done